newsfragments/validate_files.py

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python3
+
+# Towncrier silently ignores files that do not match the expected ending.
+# We use this script to ensure we catch these as errors in CI.
+
+import os
+import pathlib
+
+ALLOWED_EXTENSIONS = {
+    '.feature.rst',
+    '.bugfix.rst',
+    '.doc.rst',
+    '.removal.rst',
+    '.misc.rst',
```

Comment:
I'm not sure how common it is for `web3` to make performance improvements and hence bundle those in their own section. If you think it's worth doing that, then check out this [commit](https://github.com/ethereum/trinity/commit/d7a4b5e87fbf26e70e6fa83c1faf43def598b9f8) for the changes needed to do that.

It's worth pointing out that that sections are generally only included if they have items (meaning, even if you prepare for a perf section, it will only come up in the release notes, if it's actually used)

Reply:
I think we can put performance improvements under features for now, we haven't done very man. But, if we start to do more it would for sure be worth adding!